registry/storage/driver/swift/swift.go

```diff
@@ -590,9 +657,46 @@ func (d *driver) Delete(ctx context.Context, path string) error {
 }
 
 // URLFor returns a URL which may be used to retrieve the content stored at the given path.
-// May return an UnsupportedMethodErr in certain StorageDriver implementations.
 func (d *driver) URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error) {
-	return "", storagedriver.ErrUnsupportedMethod
+	if d.SecretKey == "" {
+		return "", storagedriver.ErrUnsupportedMethod
+	}
+
+	methodString := "GET"
+	method, ok := options["method"]
+	if ok {
+		if methodString, ok = method.(string); !ok {
+			return "", storagedriver.ErrUnsupportedMethod
+		}
+	}
+
+	if d.AccessKey != "" && methodString == "HEAD" {
+		methodString = "GET"
```

Comment:
Yes, I should probably add a comment about it. On HP Cloud (the `AccessKey` parameter is only used on HP Cloud), a "HEAD" on a temporary URL is allowed only if the signature was generated with "GET" or "POST".